### PR TITLE
Add initial unit tests and minor changes and optimizations for Query and QueryBuilder classes

### PR DESCRIPTION
**Pull request by ndvalenti**

### CODE REVIEW

File: Knittery/Network/QueryBuilder.swift [hunk 1]
```diff
@@ -9,24 +9,24 @@
...
+    var page: String?
+    var notebook: [QNotebook]
+    var craft: [QCraft]
+    var availability: [QAvailability]
+    var weight: [QWeight]
     
     init (
-        search: String? = "",
+        search: String = "",
         sort: QSort? = QSort.best,
```
Comment (ciliadesu) [inline Knittery/Network/QueryBuilder.swift]: Looks like `sort` shouldn't be optional

File: Knittery/Network/QueryBuilder.swift [hunk 2]
```diff
@@ -45,24 +56,18 @@ class QueryBuilder {
...
-            print(QueryError.invalidQuery.description)
-            // TODO: this is a really good time to implement tests, need to ensure that blank queries function the same as on the website, otherwise this needs to be less permissive
-            return nil
-        }
+        //        if let search = query.search {
+        //            valid = true
+        result += "query=" + query.search
+        //        } else {
+        //            print(QueryError.invalidQuery.description)
+        // TODO: this is a really good time to implement tests, need to ensure that blank queries function the same as on the website, otherwise this needs to be less permissive
```
Comment (ciliadesu) [inline Knittery/Network/QueryBuilder.swift]: I think now is a good time to clean up these comments and TODO

File: Knittery/Network/QueryBuilder.swift [hunk 3]
```diff
@@ -45,24 +56,18 @@ class QueryBuilder {
     static private let catQuery = "$"
```
Comment (ciliadesu) [inline Knittery/Network/QueryBuilder.swift]: I would consider renaming `catQuery` to something like `concatenation`

File: KnitteryTests/QueryTests.swift [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    
+    func testDoubleWeightsQuery() {
+        query.clear()
+        query.search = "cowl"
+        query.weight.append(.fingering)
+        query.weight.append(.jumbo)
+        let queryString = QueryBuilder.build(query)
+        
+        XCTAssertEqual(queryString, "?query=cowl$sort=best$weight=fingering%7Cjumbo")
+    }
```
Comment (ciliadesu) [inline KnitteryTests/QueryTests.swift:114]: Looks like we are missing a test to test `sort`

**Review by ciliadesu — CHANGES_REQUESTED** [review]

File: Knittery/Network/QueryBuilder.swift (hunk 1 above)
Comment (ndvalenti) [reply Knittery/Network/QueryBuilder.swift]: Removed for now, it's possible to search without specifying sort but more testing needed to see how results are ordered

File: Knittery/Network/QueryBuilder.swift (hunk 3 above)
Comment (ndvalenti) [reply Knittery/Network/QueryBuilder.swift]: Name changed to concat for clarity

File: KnitteryTests/QueryTests.swift (hunk 4 above)
Comment (ndvalenti) [reply KnitteryTests/QueryTests.swift:114]: Sort requires no standalone test as it is included in every query with testBasicInvertQuery() testing its inversion property

**Review by ciliadesu — APPROVED** [review]

**Pull request merged**